flink-connector-debezium/src/main/java/com/alibaba/ververica/cdc/debezium/DebeziumSourceFunction.java

```diff
@@ -138,7 +145,8 @@ public void open(Configuration parameters) throws Exception {
 		ThreadFactory threadFactory = new ThreadFactoryBuilder()
 			.setNameFormat("debezium-engine")
 			.build();
-		this.executor = Executors.newSingleThreadExecutor(threadFactory);
+		this.executor = Executors.newFixedThreadPool(2, threadFactory);
+		this.changeEventQueue = new LinkedBlockingQueue<>();
```

Comment:
We should set a max size for the queue to avoid OOM when the main thread is blocked.

Reply:
I wonder if it is necessary to use the `configuration` parameter here, just like: `this.changeEventQueue = new LinkedBlockingQueue<>(parameters.getInteger(ConfigOptions.key("flink.cdc.connector.change.event.queue.size").intType().defaultValue(1024)))`
Or use a constant directly in code

Reply:
Size one blocking queue is enough. This blocking queue is only used for handover data. However, I just find that blocking queue may not be sufficient here, because we need extra exception reporting. I'm wondering we may need something similar to the [`Handover`](https://github.com/apache/flink/blob/master/flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/internals/Handover.java#L50) class in kafka connector.

Reply:
Yes, it's really enough to use size one blocking queue.😎Actually the event is handled one by one, and in order to maintain the semantic of handover class, so I think we should use `ChangeEvent<SourceRecord, SourceRecord>` as the generic type. The producer put the events into the handover structure in a loop.

Reply:
I think it would be better to use `List<ChangeEvent>`, this can reduce the frequent communication between `DebeziumChangeConsumer` and `Handover`. `ConsumerRecords` is also a List of `ConsumerRecord`.

Reply:
OK, I will submit a revision later.

Reply:
done